packages/context/src/inject.ts

```diff
@@ -414,11 +420,20 @@ function resolveByFilter(
   injection: Readonly<Injection>,
   session?: ResolutionSession,
 ) {
-  const bindingFilter = injection.metadata!.bindingFilter;
+  const targetType = inspectTargetType(injection);
+  if (injection.metadata && injection.metadata.decorator === '@inject.view') {
+    if (targetType && targetType !== ContextView) {
```

Comment:
I find it weird to see this check inside `resolveByFilter` function, it feels inconsistent with the way how getter/setter targets are checked above.

I think we should we have `resolveView` function that will be dedicated to `@inject.view` only, and that would be the place where to check the target type.

Reply:
_https://github.com/strongloop/loopback-next/pull/2122/files#r248974302_

I find it weird to see this check inside `resolveByFilter` function, it feels inconsistent with the way how getter/setter targets are checked above.

I think we should we have `resolveView` function that will be dedicated to `@inject.view` only, and that would be the place where to check the target type.

Reply:
I split `resolveByFilter` into a few functions, each of which handles one `@inject` flavor.